indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -481,21 +485,24 @@ public TaskLocation getLocation()
     );
   }
 
-  private static class WaitingTask extends TaskRunnerWorkItem
+  private static class AnyTaskBuilder extends TaskRunnerWorkItem
```

Comment:
This class is not a builder. Please change to a more appropriate name.

Reply:
yeah, i was hesitant to name it this way, i added one method `withTaskState` which is kind of builder.

Reply:
It's only a builder if it has a `build()` method, or something like that, that creates an object of a different type. This is more like a normal non-builder class.

IMO, given that this is not really a builder class, the `withTaskState` method is misleadingly named. Generally, on a non-builder class, you'd expect `withX` to make a _copy_ with X changed. This is more of a setter, since it mutates the object. So a better name would be `setTaskState`. (It can still return `this`, if you want.)

Or, you could design the class to be immutable. In that case, make the field `final`, call the method `withTaskState`, and have it return a copy with a different state.

My personal preference would be to do an immutable class, because immutable classes are easier to reason about (you know they won't mutate on you after you get one).